crates/pico/tests/memo_on_struct_method.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use pico::{Database, SourceId, Storage};
+use pico_macros::{memo, Db, Source};
+
+static FIRST_LETTER_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+#[derive(Db, Default)]
+struct TestDatabase {
+    pub storage: Storage<Self>,
+}
+
+struct TestStruct;
+
+impl TestStruct {
+    #[memo(db = test_db)]
+    fn first_letter(&self, test_db: &TestDatabase, input_id: SourceId<Input>) -> char {
```

Comment:
nifty

Reply:
what if we access `self` here? That would break memoization, right?

I suppose that's true of methods on structs that implements Database, but maybe then it's clear that you are only supposed to store sources on that struct...